contrib/babelfishpg_tsql/src/session.c

```diff
@@ -51,6 +52,23 @@ get_cur_db_name(void)
 	return pstrdup(current_db_name);
 }
 
+void 
+set_cur_db_name_for_parallel_worker(const char* logical_db_name)
+{
+	int len = strlen(logical_db_name);
+
+	Assert(len <= MAX_BBF_NAMEDATALEND);
+
+	if(logical_db_name == NULL || !DbidIsValid(get_db_id(logical_db_name)))
```

Comment:
this is not helpful as you have already called strlen above. you can switch error check and assert

Reply:
Updated.